src/components/scenes/Staking/EarnScene.tsx

```diff
@@ -290,13 +288,16 @@ export const EarnScene = (props: Props) => {
       }
 
       // User backed out of the WalletListModal
-      if (walletId == null) return
+      if (walletId == null || stakePosition == null) return
+
+      dispatch({ type: 'STAKING/ADD_POLICY', walletId, stakePolicy })
+      dispatch({ type: 'STAKING/UPDATE_POSITION', walletId, stakePolicyId: stakePolicy.stakePolicyId, stakePosition })
+      dispatch({ type: 'STAKING/FINISH_LOADING', walletId })
```

Comment:
A repeated `dispatch` calls can tank performance. Maybe make a `{ type: 'STAKING/UPDATE', stakePolicy, stakePosition, walletId }` action that can configure all three values in one shot.